plugins/modules/ecs_taskdefinition.py

```diff
@@ -682,6 +686,23 @@ def register_task(self, family, task_role_arn, execution_role_arn, network_mode,
                             self.module.fail_json(msg="In awsvpc network mode, host port must be set to the same as "
                                                   "container port or not be set")
 
+            if 'linuxParameters' in container:
+                for linux_param in container.get('linuxParameters'):
+                    if linux_param == 'tmpfs':
+                        for tmpfs_param in container['linuxParameters']['tmpfs']:
+                            if 'size' in tmpfs_param:
+                                tmpfs_param['size'] = int(tmpfs_param['size'])
+
+                    for param in ('maxSwap', 'swappiness', 'sharedMemorySize'):
+                        if param in linux_param:
+                            container['linuxParameters'][param] = int(container['linuxParameters'][param])
+
+            if 'ulimits' in container:
+                for limits_mapping in container['ulimits']:
+                    for limit in ('softLimit', 'hardLimit'):
+                        if limit in limits_mapping:
+                            limits_mapping[limit] = int(limits_mapping[limit])
```

Comment:
```suggestion
                for limits_mapping in container['ulimits']:
                    for limit in ('softLimit', 'hardLimit'):
                        if limit == limits_mapping:
                            limits_mapping[limit] = int(limits_mapping[limit])
```

Reply:
@markuman `limits_mapping` is a dict, for this reason I used `if limit in limits_mapping`.